blocks/search/search.js

```diff
@@ -204,22 +232,33 @@ function searchBox(block, config) {
     }
   });
 
-  searchInput.addEventListener('input', (e) => {
-    handleSearch(e, block, config);
-  });
-
-  searchInput.addEventListener('keyup', (e) => {
+  /**
+   * Search after input into search field.
+   */
+  const debouncedHandleSearch = debounce((e) => handleSearch(e?.target, block, config), 200);
+  searchInput.addEventListener('input', debouncedHandleSearch);
+
+  /**
+   * Kick off search if search field already has a value when it gains focus.
+   */
+  searchInput.addEventListener('focus', (e) => handleSearch(e?.target, block, config));
+
+  /**
+   * Handle escape being pressed on input or when focused on a result.
+   */
+  block.addEventListener('keyup', (e) => {
+    // Collapse and clear search after pressing Escape.
     if (e.code === 'Escape') {
       box.classList.remove('search__box--expanded');
       toggleButton.toggleAttribute('aria-expanded');
       searchInput.value = '';
```

Comment:
UX and functionality question here; should this value not clear when toggling the search button to hide the search after typing a value in it? Note; clicking somewhere else on the page rather than the toggle currently does not clear it.